--- a/cmake/Geant4.cmake
+++ b/cmake/Geant4.cmake
@@ -150,7 +150,12 @@
 string(FIND ${Geant4_CXX_FLAGS} "c++17" _G4CXX17FOUND)
 string(FIND ${Geant4_CXX_FLAGS} "c++1z" _G4CXX17FOUND2)
 string(FIND ${Geant4_CXX_FLAGS} "c++20" _G4CXX20FOUND)
-if (_G4CXX20FOUND STRGREATER -1)
+string(FIND ${Geant4_CXX_FLAGS} "c++23" _G4CXX23FOUND)
+if (_G4CXX23FOUND STRGREATER -1)
+  message(STATUS "Geant4 compiled with cxx23 -> changing to C++23 for BDSIM")
+  set(CMAKE_CXX_STANDARD 23)
+  set(CMAKE_CXX_STANDARD_REQUIRED ON)
+elseif (_G4CXX20FOUND STRGREATER -1)
   message(STATUS "Geant4 compiled with cxx20 -> changing to C++20 for BDSIM")
   set(CMAKE_CXX_STANDARD 20)
   set(CMAKE_CXX_STANDARD_REQUIRED ON)
