--- a/lib/library.cpp
+++ b/lib/library.cpp
@@ -215,5 +215,2 @@
             std::list<std::string> cfgfolders;
-    #ifdef FILESDIR
-            cfgfolders.emplace_back(FILESDIR "/cfg");
-    #endif
             if (exename) {
@@ -222,2 +219,3 @@
-                cfgfolders.push_back(std::move(exepath));
+                cfgfolders.push_back(exepath);
+                cfgfolders.push_back(std::move(exepath) + "../share/Cppcheck/cfg");
             }
