--- a/src/OutputPolicyHepMC3.h
+++ b/src/OutputPolicyHepMC3.h
@@ -29,7 +29,10 @@
   HepMC3::WriterAscii* _writer = 0;
   HepMC3::GenEvent _event;
 
-protected:
+// 10.11.2021 temporary workaround provided by Andrii Verbitskii switching from private to public 
+// // to deal with compilation problem with gcc11
+//
+public:
   // --------------  test observables
   template <typename T>
   struct Stat {
--- a/src/OutputPolicyHepMC.h
+++ b/src/OutputPolicyHepMC.h
@@ -34,7 +34,10 @@
 	     HepMC::GenEvent> hepevt;
   HepMC::IO_GenEvent* ascii_out;
 
- protected:
+// 10.11.2021 temporary workaround provided by Andrii Verbitskii switching from private to public 
+// to deal with compilation problem with gcc11
+
+ public:
   // --------------  test observables
   template <typename T>
   struct Stat {
--- a/./src/CRMCtrapfpe.c
+++ b/./src/CRMCtrapfpe.c
@@ -1,9 +1,9 @@
 #ifndef __INCLUDE_TRAPFE_H_
 #define __INCLUDE_TRAPFE_H_
 
-// skip compilation on MacOSX platform
+// skip compilation on MacOSX or ARM platform
 #ifndef __APPLE__
-
+#ifndef __aarch64__
 //
 // from: http://www.fortran-2000.com/ArnaudRecipes/CompilerTricks.html
 //
@@ -36,5 +36,6 @@
 }
 
 #endif
+#endif
 
 #endif
--- a/CMakeLists.txt.org
+++ b/CMakeLists.txt
@@ -185,1 +185,1 @@
-FIND_PACKAGE (HepMC3 COMPONENTS HepMC3)
+FIND_PACKAGE (HepMC3)
@@ -507,6 +507,8 @@
   src/epos.incsem
   DESTINATION include/crmc)
 INSTALL (FILES ${CRMC_HEADERS} DESTINATION include/crmc)
+FILE(GLOB all_headers ${CMAKE_SOURCE_DIR}/src/*.h ${CMAKE_BINARY_DIR}/src/*.h)
+INSTALL (FILES ${all_headers} DESTINATION include/crmc) 
 
 # enable dashboard scripting
 include (CTest)
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -171,5 +171,5 @@
 # SET(Boost_DEBUG TRUE)
 FIND_PACKAGE (Boost 1.35 REQUIRED
-  COMPONENTS filesystem iostreams system program_options)
+  COMPONENTS filesystem iostreams program_options)
 
 FIND_PACKAGE (HepMC COMPONENTS HepMC HepMCfio)
