--- a/configure
+++ b/configure
@@ -178,7 +178,8 @@
   esac
 done
 
-test "$CONF_OS" = Linux -a "$CONF_BITS" \!= 32 && CONF_LIBDIRSUFFIX=64
+## patch to avoid lib64 directory for Ubuntu/Debian
+test "$CONF_OS" = Linux -a "$CONF_BITS" \!= 32 && test ! -f /etc/debian_version && CONF_LIBDIRSUFFIX=64
 
 CONF_M=${CONF_BITS:+-m$CONF_BITS}
 
--- a/makefile.in
+++ b/makefile.in
@@ -41,8 +41,6 @@
          $(BLD)/PDG.h "$(INCLUDEDIR)"
        strip $(BLD)/FeynHiggs$(EXE) $(BLD)/table$(EXE)
        cp -p $(BLD)/FeynHiggs$(EXE) $(BLD)/table$(EXE) $(BLD)/fcc "$(BINDIR)"
-       -strip $(BLD)/MFeynHiggs$(EXE)
-       -cp -p $(BLD)/MFeynHiggs$(EXE) "$(BINDIR)"
 
 force: $(BLD)/version.h

