--- a/src/core/lib/gprpp/per_cpu.h
+++ b/src/core/lib/gprpp/per_cpu.h
@@ -87,1 +87,1 @@
-  explicit PerCpu(PerCpuOptions options) : shards_(options.Shards()) {}
+  explicit PerCpu(PerCpuOptions options) : shards_(options.Shards()), data_{new T[shards_]} {}
@@ -99 +99 @@
-  std::unique_ptr<T[]> data_{new T[shards_]};
+  std::unique_ptr<T[]> data_;
--- src/core/lib/promise/detail/basic_seq.h
+++ src/core/lib/promise/detail/basic_seq.h
@@ -99,8 +99,7 @@ class BasicSeqIter {
           }
           cur_ = next;
           state_.~State();
-          Construct(&state_,
-                    Traits::template CallSeqFactory(f_, *cur_, std::move(arg)));
+          Construct(&state_, Traits::CallSeqFactory(f_, *cur_, std::move(arg)));
           return PollNonEmpty();
         });
   }
--- a/cmake/systemd.cmake
+++ b/cmake/systemd.cmake
@@ -15,6 +15,6 @@
-find_package(systemd)
-if(TARGET systemd)
-  set(_gRPC_SYSTEMD_LIBRARIES systemd ${SYSTEMD_LINK_LIBRARIES})
-  add_definitions(-DHAVE_LIBSYSTEMD)
-endif()
-set(_gRPC_FIND_SYSTEMD "if(NOT systemd_FOUND)\n  find_package(systemd)\nendif()")
+# find_package(systemd)
+# if(TARGET systemd)
+#   set(_gRPC_SYSTEMD_LIBRARIES systemd ${SYSTEMD_LINK_LIBRARIES})
+#   add_definitions(-DHAVE_LIBSYSTEMD)
+# endif()
+# set(_gRPC_FIND_SYSTEMD "if(NOT systemd_FOUND)\n  find_package(systemd)\nendif()")
