--- a/examples/hepmcreader/reader.cc
+++ b/examples/hepmcreader/reader.cc
@@ -172,7 +172,7 @@
   for (vector<baseAnalysis*>::const_iterator i(analysis.begin()); i != analysis.end(); ++i) {
     (*i)->Finalize(f);
   }
-
+f->Close();
   // clean up analysis modules and configuration
   for (vector<baseAnalysis*>::const_iterator i(analysis.begin()); i != analysis.end(); ++i) {
     delete (*i);
--- a/config.mk
+++ b/config.mk
@@ -10,16 +10,16 @@
 #              -I$(HepPDTdir)/include -I$(ROOTSYS)/include
 INCLUDES    = $(INCLUDEH) $(INCLUDEF)
 
-ROOTGLIBS   = `root-config --glibs`
-ROOTCFLAGS  = `root-config --cflags` -Wno-long-long
+ROOTGLIBS   = `$(ROOTSYS)/bin/root-config --glibs`
+ROOTCFLAGS  = `$(ROOTSYS)/bin/root-config --cflags` -Wno-long-long
 
 #CLHEP = -L$(CLHEPdir)/lib -lCLHEP    # causes problems in HepMCAnalysis_i interface!
-CLHEPlib    = 
+CLHEPlib    = -L$(CLHEPdir)/lib -lCLHEP
 
 # common libraries
 HepPDTlib   = 
 #HepPDTlib   = -L$(HepPDTdir)/lib -lHepPDT -lHepPID
-HepMClib    = -L$(HepMCdir)/lib -lHepMC -lHepMCfio
+HepMClib    = -L$(HepMCdir)/lib -lHepMC
 
 FastJetlib  = `$(FastJetdir)/bin/fastjet-config --libs --plugins=yes --rpath=no`
 # output directories
--- a/Makefile
+++ b/Makefile
@@ -28,7 +28,7 @@
 	@mkdir -p $(LIBDIR) 
 	$(CXX) $(CXXFLAGS) $(INCLUDES) \
 	-shared -o $(LIBDIR)/libHepMCAnalysis.so $(OBJECTS) \
-	$(HepMClib) $(HepPDTlib) $(FastJetlib) \
+	$(HepMClib) $(HepPDTlib) $(FastJetlib) $(CLHEPlib) \
 	$(ROOTGLIBS) $(LINK_LIBS)


--- a/src/baseAnalysis.cc
+++ b/src/baseAnalysis.cc
@@ -281,8 +281,8 @@
     double pe = (*p)->momentum().e();
     
       // remove garbage 
-    if ( isnan( px ) || isnan( py ) || isnan( pz ) || isnan( pe ) || 
-         isinf( px ) || isinf( py ) || isinf( pz ) || isinf( pe ) ) {
+    if ( std::isnan( px ) || std::isnan( py ) || std::isnan( pz ) || std::isnan( pe ) || 
+         std::isinf( px ) || std::isinf( py ) || std::isinf( pz ) || std::isinf( pe ) ) {
       cout << "baseAnalysis: NaN (Not A Number) found in the event record! Particle will be skipped." << endl;
       ++m_NanNum;
       continue;
