--- a/CMake/HighFiveTargetDeps.cmake
+++ b/CMake/HighFiveTargetDeps.cmake
@@ -53,5 +53,5 @@
       set(Boost_NO_BOOST_CMAKE TRUE)
     endif()
-    find_package(Boost REQUIRED COMPONENTS system serialization)
+    find_package(Boost REQUIRED COMPONENTS serialization)
     # Dont use imported targets yet, not avail before cmake 3.5
     target_include_directories(libdeps SYSTEM INTERFACE ${Boost_INCLUDE_DIR})
