--- ./horovod/torch/CMakeLists.txt
+++ ./horovod/torch/CMakeLists.txt
@@ -63,7 +63,9 @@ list(APPEND PYTORCH_LINKER_LIBS ${Pytorch_LIBRARIES})
 parse_version(${Pytorch_VERSION} VERSION_DEC)
 add_definitions(-DPYTORCH_VERSION=${VERSION_DEC} -DTORCH_API_INCLUDE_EXTENSION_H=1)
 set(Pytorch_CXX11 ${Pytorch_CXX11} PARENT_SCOPE)
-if(NOT Pytorch_VERSION VERSION_LESS "1.5.0")
+if(NOT Pytorch_VERSION VERSION_LESS "2.1.0")
+    set(CMAKE_CXX_STANDARD 17)
+elseif(NOT Pytorch_VERSION VERSION_LESS "1.5.0")
     set(CMAKE_CXX_STANDARD 14)
 endif()
 
--- a/./third_party/gloo/gloo/rendezvous/store.cc
+++ b/./third_party/gloo/gloo/rendezvous/store.cc
@@ -14 +14 @@
-constexpr std::chrono::milliseconds Store::kDefaultTimeout;
+  //constexpr std::chrono::milliseconds Store::kDefaultTimeout;
--- a/horovod/common/common.h
+++ b/horovod/common/common.h
@@ -21,6 +21,7 @@
 #include <functional>
 #include <memory>
 #include <string>
+#include <stdexcept>
 #include <unordered_map>
 #include <unordered_set>
 
--- a/horovod/common/group_table.h
+++ b/horovod/common/group_table.h
@@ -17,6 +17,7 @@
 #include <queue>
 #include <unordered_map>
 #include <vector>
+#include <string>
 
 #ifndef HOROVOD_GROUP_TABLE_H
 #define HOROVOD_GROUP_TABLE_H

--- a/third_party/flatbuffers/include/flatbuffers/stl_emulation.h
+++ b/third_party/flatbuffers/include/flatbuffers/stl_emulation.h
@@ -625,7 +625,7 @@
  private:
   // This is a naive implementation with 'count_' member even if (Extent != dynamic_extent).
   pointer const data_;
-  const size_type count_;
+  size_type count_;
 };
 
  #if !defined(FLATBUFFERS_SPAN_MINIMAL)
--- a/./horovod/runner/js_run.py
+++ b/./horovod/runner/js_run.py
@@ -23 +23,2 @@
-from distutils.spawn import find_executable
+#from distutils.spawn import find_executable
+from shutil import which as find_executable
--- a/third_party/gloo/gloo/types.h
+++ b/third_party/gloo/gloo/types.h
@@ -6,4 +6,5 @@
 
 #include <iostream>
+#include <cstdint>
 
 #ifdef __CUDA_ARCH__
