--- a/src/profile.cc
+++ b/src/profile.cc
@@ -32 +32 @@
-HIDDEN void             (*igprof_abort)(void) throw () = &abort;
+HIDDEN void             (*igprof_abort)(void) = &abort;
--- a/src/profile.h
+++ b/src/profile.h
@@ -14 +14 @@
-typedef void IgProfAbortFunc (void) __attribute__((noreturn));
+typedef void IgProfAbortFunc (void);
@@ -24,1 +24,1 @@
-extern void             (*igprof_abort) (void) throw ();
+extern void             (*igprof_abort) (void) ;
