--- a/src/Makefile
+++ b/src/Makefile
@@ -19,7 +19,7 @@
 $(LIB_SHARED):  $(objects)
 	mkdir -p $(LIBDIR_SHARED)
 	@echo creating the shared library $@
-	$(FC) $(LDFLAGSSHARED) $(objects) -o $@ -shared -Wl,-soname,$(notdir $@)
+	$(FC) $(LDFLAGSSHARED) $(objects) $(LIBS_HERWIG) -o $@ -shared 
 
 $(LIB_ARCHIVE) : $(objects)
 	mkdir -p $(LIBDIR_ARCHIVE)
--- a/configure
+++ b/configure
@@ -63,8 +63,8 @@
 #
 # export FFLAGS_OPT="-O2 -Wuninitialized"
 # -Wuninitialized outputs many irrelevant warnings and therefore obscurs
-export FFLAGS_OPT="-O2"
-export FFLAGS_DBG=-g
+export FFLAGS_OPT="-O2 -std=legacy"
+export FFLAGS_DBG="-g -std=legacy"
 export CFLAGS_OPT=-O2
 export CFLAGS_DBG=-g
 export CXXFLAGS_OPT="-O2 -ansi -pedantic -W -Wall"
