--- a/include/log4cpp/config.h
+++ b/include/log4cpp/config.h
@@ -0,0 +1,3 @@
+#pragma once
+#define LOG4CPP_HAVE_UNISTD_H 1
+#define LOG4CPP_HAVE_SSTREAM 1
--- a/include/log4cpp/DailyRollingFileAppender.hh
+++ b/include/log4cpp/DailyRollingFileAppender.hh
@@ -13,0 +14 @@
+#include <time.h>
--- a/src/snprintf.c
+++ b/src/snprintf.c
@@ -364 +364 @@
-  { register size_t nn = (size_t)(n); \
+  { size_t nn = (size_t)(n); \
@@ -367 +367 @@
-      register char *dd; register const char *ss; \
+      char *dd; const char *ss; \
@@ -371 +371 @@
-  { register size_t nn = (size_t)(n); \
+  { size_t nn = (size_t)(n); \
@@ -374,1 +374,1 @@
-      register char *dd; register const int cc=(int)(c); \
+      char *dd; const int cc=(int)(c); \
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -93,1 +92,0 @@
-  PATTERN "config.h" EXCLUDE
