--- CMakeLists.txt~	2024-05-17 11:29:32.000000000 +0200
+++ CMakeLists.txt	2025-04-23 10:15:45.976631157 +0200
@@ -508,5 +508,5 @@
 if(ONNX_USE_LITE_PROTO)
   if(TARGET protobuf::libprotobuf-lite)
-    target_link_libraries(onnx_proto PUBLIC protobuf::libprotobuf-lite PRIVATE ${protobuf_ABSL_USED_TARGETS})
+    target_link_libraries(onnx_proto PUBLIC protobuf::libprotobuf-lite PUBLIC ${protobuf_ABSL_USED_TARGETS})
   else()
     target_link_libraries(onnx_proto PUBLIC ${PROTOBUF_LITE_LIBRARIES})
@@ -514,5 +514,5 @@
 else()
   if(TARGET protobuf::libprotobuf)
-    target_link_libraries(onnx_proto PUBLIC protobuf::libprotobuf PRIVATE ${protobuf_ABSL_USED_TARGETS})
+    target_link_libraries(onnx_proto PUBLIC protobuf::libprotobuf PUBLIC ${protobuf_ABSL_USED_TARGETS})
   else()
     target_link_libraries(onnx_proto PUBLIC ${PROTOBUF_LIBRARIES})
--- requirements.txt	2025-07-25 14:43:15
+++ requirements.txt	2025-07-25 14:41:02
@@ -2 +2,2 @@
 protobuf>=3.20.2
+cmake<4.0
--- pyproject.toml	2025-07-25 14:51:10
+++ pyproject.toml	2025-07-25 14:50:47
@@ -3,3 +3,3 @@
 [build-system]
-requires = ["setuptools>=64", "protobuf>=3.20.2", "cmake"]
+requires = ["setuptools>=64", "protobuf>=3.20.2", "cmake<4.0"]
 build-backend = "setuptools.build_meta"
