--- cmake/protobuf.pc.cmake~	2025-09-08 13:39:31.000000000 +0200
+++ cmake/protobuf.pc.cmake	2025-09-09 13:05:01.973434559 +0200
@@ -10 +10 @@
-Libs: -L${libdir} -lprotobuf @CMAKE_THREAD_LIBS_INIT@
+Libs: -L${libdir} -lprotobuf@protobuf_LIBRARY_SUFFIX@ @CMAKE_THREAD_LIBS_INIT@
--- cmake/install.cmake~	2025-09-08 13:39:31.000000000 +0200
+++ cmake/install.cmake	2025-09-09 13:04:38.677359778 +0200
@@ -22,0 +23,4 @@
+if(CMAKE_BUILD_TYPE STREQUAL Debug)
+  set(protobuf_LIBRARY_SUFFIX d)
+endif()
+
