--- Makefile.am.res	2026-01-12 15:31:31.414888491 +0100
+++ Makefile.am	2026-01-12 15:35:17.168492505 +0100
@@ -4,8 +4,8 @@
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = qd.pc
 
-BUILT_SOURCES = docs/qd.pdf
-dist_doc_DATA = README README.md docs/qd.pdf
+BUILT_SOURCES = 
+dist_doc_DATA = README README.md
 dist_noinst_DATA = docs/Makefile \
   docs/qd.tex docs/qd.bib \
   docs/nine-two-sum.eps docs/nine-two-sum.fig \
