--- a/qtbase/src/platformsupport/input/xkbcommon/qxkbcommon.cpp
+++ b/qtbase/src/platformsupport/input/xkbcommon/qxkbcommon.cpp
@@ -275,0 +276 @@
+#ifdef XKB_KEY_dead_lowline
@@ -280 +281 @@
-
+#endif
--- a/qtlocation/src/3rdparty/mapbox-gl-native/src/mbgl/gl/stencil_mode.hpp
+++ b/qtlocation/src/3rdparty/mapbox-gl-native/src/mbgl/gl/stencil_mode.hpp
@@ -2,0 +3,2 @@
+#include <cstdint>
+
--- a/qtlocation/src/3rdparty/mapbox-gl-native/include/mbgl/util/geometry.hpp
+++ b/qtlocation/src/3rdparty/mapbox-gl-native/include/mbgl/util/geometry.hpp
@@ -2,0 +3,2 @@
+#include <cstdint>
+
--- a/qtlocation/src/3rdparty/mapbox-gl-native/include/mbgl/util/string.hpp
+++ b/qtlocation/src/3rdparty/mapbox-gl-native/include/mbgl/util/string.hpp
@@ -2,0 +3 @@
+#include <cstdint>
--- a/qtlocation/src/3rdparty/mapbox-gl-native/deps/rapidjson/1.1.0/include/rapidjson/document.h
+++ b/qtlocation/src/3rdparty/mapbox-gl-native/deps/rapidjson/1.1.0/include/rapidjson/document.h
@@ -316,8 +316,6 @@
 
     GenericStringRef(const GenericStringRef& rhs) : s(rhs.s), length(rhs.length) {}
 
-    GenericStringRef& operator=(const GenericStringRef& rhs) { s = rhs.s; length = rhs.length; }
-
     //! implicit conversion to plain CharType pointer
     operator const Ch *() const { return s; }
 
@@ -328,6 +326,8 @@
     //! Disallow construction from non-const array
     template<SizeType N>
     GenericStringRef(CharType (&str)[N]) /* = delete */;
+    //! Copy assignment operator not permitted - immutable type
+    GenericStringRef& operator=(const GenericStringRef& rhs) /* = delete */;
 };
 
 //! Mark a character pointer as constant string
