--- a/src/Tools/binreloc.c
+++ b/src/Tools/binreloc.c
@@ -215,7 +215,10 @@
 			break;
 
 		/* Sanity check. */
-		if (strstr (line, " r-xp ") == NULL || strchr (line, '/') == NULL)
+                /* With binutils > 2.30 (const void *) "" is placed in "r--p" area of memory
+                   In general, for the mechanism of detection it doesn't really matter where your symbol is,
+                   therefore, check for strstr (line, " r-xp ") == NULL is removed */
+		if (strchr (line, '/') == NULL)
 			continue;
 
 		/* Parse line. */

--- a/configure.ac
+++ b/configure.ac
@@ -215,7 +215,7 @@
   if test "$fjmajor" -lt 3; then
     AC_MSG_ERROR([FastJet version 3.0.0 or later is required])
   fi
-  FASTJETLIBADD="$($FJCONFIG --plugins --shared --libs | sed 's/\S*\/gcc\/\S*//g')"
+  FASTJETLIBADD="$($FJCONFIG --plugins --shared --libs)"
 else
   FASTJETLIBADD="-L$FASTJETLIBPATH -l$FASTJETLIBNAME"
   FASTJETLIBADD="$FASTJETLIBADD -lSISConePlugin -lsiscone -lsiscone_spherical"
