--- a/signalhandling.h
+++ b/signalhandling.h
@@ -61,7 +61,7 @@
 void initSignalHandler(int nointeract);
 void blockSignals(int nointeract);
 void initSignalHandlerCounted(int nointeract);
-void blockSignalsCounted();
+void blockSignalsCounted(int nointeract);
 void deferSignalHandling();
 void resumeSignalHandling();
 
--- a/execute.h
+++ b/execute.h
@@ -80,7 +80,7 @@
 #include "general.h"
 #include "expression.h"
 
-extern int miniyyparse();
+extern int miniyyparse(void *myScanner);
 extern void startBuffer(char *str);
 extern void endBuffer(void);
 
