--- a/src/Makefile
+++ b/src/Makefile
@@ -1,7 +1,8 @@
-COPTS = -O -DDROP_CGAL
-INCLUDES = -I../include -I$(shell lhapdf-config --incdir)
+COPTS = -O -DDROP_CGAL $(CXXFLAGS)
+INCLUDES = -I../include -I$(shell $(LHAPDF_HOME)/bin/lhapdf-config --incdir) -I$(BOOST_INCLUDE)
+
 OBJECTS = SysCalc.o tinyxml2.o alfas_functions.o
-LIBS = $(shell lhapdf-config --ldflags)
+LIBS = $(shell lhapdf-config --ldflags) -L$(BOOST_HOME)/lib
 
 .cc.o:
 	$(CXX) $(COPTS) $(INCLUDES) -c $<
@@ -18,7 +19,7 @@
 	$(CXX) -o $@ $^ $(LIBS) -lgfortran
 
 alfas_functions.o: alfas_functions.f
-	gfortran -c alfas_functions.f
+	gfortran -fPIE -c alfas_functions.f
 
 clean:
 	rm -f ../sys_calc *.o
\ No newline at end of file
--- a/include/tinyxml2.h
+++ b/include/tinyxml2.h
@@ -191 +191 @@
-    DynArray< T, INIT >() {
+    DynArray() {
