--- a/./tensorflow_probability/python/__init__.py
+++ b/./tensorflow_probability/python/__init__.py
@@ -49,2 +49,2 @@
-  import distutils.version
-
+#  import distutils.version
+  from looseversion import LooseVersion
@@ -57,2 +57,4 @@
-  if (distutils.version.LooseVersion(tf.__version__) <
-      distutils.version.LooseVersion(required_tensorflow_version)):
+#  if (distutils.version.LooseVersion(tf.__version__) <
+#      distutils.version.LooseVersion(required_tensorflow_version)):
+  if (LooseVersion(tf.__version__) <
+      LooseVersion(required_tensorflow_version)):
