--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -12,0 +13 @@
+  add_definitions(WITH_MY_CUSTOM_DEF)
@@ -34,0 +36,2 @@
+include(GNUInstallDirs)
+
@@ -36 +39 @@
-  "$<BUILD_INTERFACE:${HEADERS}>"
+  $<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/csrc>
@@ -39 +41,0 @@
-include(GNUInstallDirs)
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -38 +38 @@
-target_include_directories(${PROJECT_NAME} INTERFACE
+target_include_directories(${PROJECT_NAME} PUBLIC
--- a/setup.py
+++ b/setup.py
@@ -49 +49,2 @@
-            extra_compile_args['cxx'] += ['-Wno-sign-compare']
+            pass
+            #extra_compile_args['cxx'] += ['-Wno-sign-compare']
@@ -100 +101 @@
-            include_dirs=[extensions_dir],
+            include_dirs=[os.path.join(os.getcwd(), extensions_dir)],
--- a/setup.py
+++ b/setup.py
@@ -11,2 +11,6 @@
-from torch.utils.cpp_extension import (CUDA_HOME, BuildExtension, CppExtension,
-                                       CUDAExtension)
+from torch.utils.cpp_extension import (
+    CUDA_HOME,
+    BuildExtension,
+    CppExtension,
+    CUDAExtension,
+)
@@ -20 +23,0 @@
-
@@ -49,2 +52 @@
-            pass
-            #extra_compile_args['cxx'] += ['-Wno-sign-compare']
+            extra_compile_args['cxx'] += ['-Wno-sign-compare']
@@ -76,2 +77,0 @@
-            extra_compile_args['nvcc'] = nvcc_flags
-
@@ -85,0 +86,2 @@
+            extra_compile_args['nvcc'] = nvcc_flags
+
@@ -148,2 +150 @@
-        'build_ext':
-        BuildExtension.with_options(no_python_abi_suffix=True, use_ninja=False)
+        'build_ext': BuildExtension.with_options(no_python_abi_suffix=True)
